ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilderV1.java

```diff
@@ -0,0 +1,293 @@
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.state.KeyValueStore;
+
+final class SourceBuilderV1 extends SourceBuilderBase {
+
+  private final static SourceBuilderV1 instance;
+
+  static {
+    instance = new SourceBuilderV1();
+  }
+
+  private SourceBuilderV1() {
+  }
+
+  public static SourceBuilderV1 instance() {
+    return instance;
+  }
+
+  public KStreamHolder<GenericKey> buildStream(
+      final RuntimeBuildContext buildContext,
+      final StreamSource source,
+      final ConsumedFactory consumedFactory
+  ) {
+    final PhysicalSchema physicalSchema = getPhysicalSchema(source);
+
+    final Serde<GenericRow> valueSerde = getValueSerde(buildContext, source, physicalSchema);
+
+    final Serde<GenericKey> keySerde = buildKeySerde(
+        source,
+        physicalSchema,
+        buildContext
+    );
+
+    final Consumed<GenericKey, GenericRow> consumed = buildSourceConsumed(
+        source,
+        keySerde,
+        valueSerde,
+        AutoOffsetReset.LATEST,
+        buildContext,
+        consumedFactory
+    );
+
+    final KStream<GenericKey, GenericRow> kstream = buildKStream(
+        source,
+        buildContext,
+        consumed,
+        nonWindowedKeyGenerator(source.getSourceSchema())
+    );
+
+    return new KStreamHolder<>(
+        kstream,
+        buildSchema(source, false),
+        ExecutionKeyFactory.unwindowed(buildContext)
+    );
+  }
+
+  public KStreamHolder<Windowed<GenericKey>> buildWindowedStream(
+      final RuntimeBuildContext buildContext,
+      final WindowedStreamSource source,
+      final ConsumedFactory consumedFactory
+  ) {
+    final PhysicalSchema physicalSchema = getPhysicalSchema(source);
+
+    final Serde<GenericRow> valueSerde = getValueSerde(buildContext, source, physicalSchema);
+
+    final WindowInfo windowInfo = source.getWindowInfo();
+    final Serde<Windowed<GenericKey>> keySerde = buildWindowedKeySerde(
+        source,
+        physicalSchema,
+        buildContext,
+        windowInfo
+    );
+
+    final Consumed<Windowed<GenericKey>, GenericRow> consumed = buildSourceConsumed(
+        source,
+        keySerde,
+        valueSerde,
+        AutoOffsetReset.LATEST,
+        buildContext,
+        consumedFactory
+    );
+
+    final KStream<Windowed<GenericKey>, GenericRow> kstream = buildKStream(
+        source,
+        buildContext,
+        consumed,
+        windowedKeyGenerator(source.getSourceSchema())
+    );
+
+    return new KStreamHolder<>(
+        kstream,
+        buildSchema(source, true),
+        ExecutionKeyFactory.windowed(buildContext, windowInfo)
+    );
+  }
+
+  @Override
+  public Materialized<GenericKey, GenericRow, KeyValueStore<Bytes, byte[]>> buildTableMaterialized(
+      final SourceStep<KTableHolder<GenericKey>> source,
+      final RuntimeBuildContext buildContext,
+      final MaterializedFactory materializedFactory,
+      final Serde<GenericKey> sourceKeySerde,
```

Comment:
nit: why was this renamed from `keySerde` to `sourceKeySerde` in the latest commit? The name is now inconsistent with `valueSerde` on the line below. (Same for `buildWindowedTableMaterialized()` below.)